engine/src/test/java/org/camunda/bpm/engine/test/api/optimize/GetHistoricOperationLogsForOptimizeTest.java

```diff
@@ -98,36 +106,395 @@ public void cleanUp() {
     for (User user : identityService.createUserQuery().list()) {
       identityService.deleteUser(user.getId());
     }
-    for (Group group : identityService.createGroupQuery().list()) {
-      identityService.deleteGroup(group.getId());
-    }
-    for (Authorization authorization : authorizationService.createAuthorizationQuery().list()) {
-      authorizationService.deleteAuthorization(authorization.getId());
-    }
     ClockUtil.reset();
     identityService.clearAuthentication();
+    helper.removeAllRunningAndHistoricBatches();
   }
 
   @Test
-  public void getHistoricUserOperationLogs() {
-     // given
+  public void getHistoricUserOperationLogs_suspendProcessInstanceByProcessInstanceId() {
+    // given
+    BpmnModelInstance simpleDefinition = Bpmn.createExecutableProcess("process")
+      .startEvent("startEvent")
+      .userTask("userTask")
+      .name("task")
+      .endEvent("endEvent")
+      .done();
+    testHelper.deploy(simpleDefinition);
+    final ProcessInstance processInstance = runtimeService.startProcessInstanceByKey("process");
+
+    runtimeService.suspendProcessInstanceById(processInstance.getProcessInstanceId());
+    runtimeService.activateProcessInstanceById(processInstance.getProcessInstanceId());
```

Comment:
Most of the tests execute multiple operations and then rely on the ordering of op log entries by timestamp; in order to avoid flaky tests, we should make sure that time actually elapses between the operations with respect to the database's precision (milliseconds in most cases; seconds for MySQL); this can be done by using `ClockUtil#offset` (similar to how it is done in `#occurredAfterParameterWorks`).

Reply:
Oh yeah, good point :-)